lib/misc.coffee

```diff
@@ -27,33 +26,104 @@ relativeFromAbsolute = (position, rte)->
 # relative position in word index and offset
 #
 # @param [Option] relative the position
-# @param [Rte] rte an rte instance
+# @param [Y.RichText] richText an richText instance
 # @return [Integer] the absolute index
-absoluteFromRelative = (index, offset, rte) ->
+absoluteFromRelative = (index, offset, richText) ->
   absolute = offset
   if index > 0
     for i in [0..(index-1)]
-      absolute += rte.getWord(i).word.length
+      absolute += richText.getWord(i).word.length
 
   absolute
 
+# Initialize an Yjs list and add commont array operations
+# over it
+customList = (Operation, self) ->
+  ret = new Operation.ListManager(@).execute()
+
+  # Another name for delete is slice
+  # TODO: remove all occurences of slice!
+  ret.slice = (start, end) ->
+    @delete start, end
+  ret.splice = (start, end, insert...) ->
+    @delete start, end
+    @insert start, insert
+  ret.indexOf = (element) ->
+    @val().indexOf(element)
+
+  ret.observe = (e) ->
+    console.log e
+
+  return ret
+
+# a basic class with generic getter / setter funciton
+class BaseClass
+  constructor: ->
+
+  # Try to find the property in @_model, else return the
+  # own property
+  _get: (prop) ->
+    if @hasOwnProperty(prop)
+      @[prop]
+    else
+      @_model.val(prop)
+  # Try to set the property in @_model, else set the
+  # own property
+  _set: (prop, val) ->
+    if @hasOwnProperty(prop)
+      @[prop] = val
+    else
+      @_model.val(prop, val)
+
+
 # Simple class that contains a word and links to the selections pointing
 # to it
 #
-class Word
-  # Attribute containing the string
-  @word = ''
-  # Selections that have this word as left bound
-  @left = []
-  # Selections that have this word as right bound
-  @right = []
+class Word extends BaseClass
+    # Attribute containing the string
+
+  _name: "Word"
+
+  _getModel: (Y, Operation) ->
+    if @_model == null
+      @_model = new Operation.MapManager(@).execute()
+      # create left and right list of selections
+      left = new Operation.MapManager(@).execute()
+      left.insert 0, @left
+
+      right = new Operation.MapManager(@).execute()
+      right.insert 0, @right
+
+      # extend the lists
+      extend @_model, customList
+      extend left, customList
+      extend right, customList
+
+      @_model.val("left", left)
+      @_model.val("right", right)
+      @_model.val("word", @word)
+      @_model.val("richText", @richText)
+
+      delete @left
+      delete @right
+      delete @richText
```

Comment:
you forgot to delete word, right? It is also a good idea to just call _setModel here